### PR TITLE
DFPL-1202: Data Breach removal of placement

### PR DESCRIPTION
**Pull request by AlistairEdwinOsborne**
https://tools.hmcts.net/jira/browse/DFPL-1202
https://tools.hmcts.net/jira/browse/DFPL-1195
https://tools.hmcts.net/jira/browse/DFPL-1204
https://tools.hmcts.net/jira/browse/DFPL-1218

Remove placement(s) from applications

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java [hunk 1]
```diff
@@ -141,146 +134,6 @@ void shouldSetCaseStateToCaseManagement() {
...
-                buildCaseDetails(caseData, migrationId)
-            );
-
-            CaseData responseData = extractCaseData(response);
-            List<Element<Child>> expectedChildren = List.of(expectedChild1, expectedChild2);
-
-            assertThat(responseData.getAllChildren()).isEqualTo(expectedChildren);
-        }
-    }
-
```
Comment (qasimnawaz-moj) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java:283]: Do we need tests for 1202?

Comment (qasimnawaz-moj) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java:283]: Just saw the commit message, all good to go!

**Review by qasimnawaz-moj — DISMISSED** [review]

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java (hunk 1 above)
Comment (AlistairEdwinOsborne) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java:283]: nah 1161 is already testing the exact same thing

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java [hunk 2]
```diff
@@ -60,147 +59,276 @@ void shouldThrowExceptionWhenMigrationNotMappedForMigrationID() {
...
-            .build());
-        final Element<Child> expectedChild2 = element(child2Id, Child.builder()
-            .party(ChildParty.builder()
-                .firstName("Fred")
-                .lastName("Frederson")
-                .completionDate(extensionDate)
-                .extensionReason(TIMETABLE_FOR_PROCEEDINGS)
-                .build())
-            .build());
+    class Dfpl1064 {
```
Comment (so99y) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java:62]: wondering why DFPL-1064 test cases is here.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java [hunk 3]
```diff
@@ -60,147 +59,276 @@ void shouldThrowExceptionWhenMigrationNotMappedForMigrationID() {
...
+
+    private CaseDetails buildCaseDetails(CaseData caseData, String migrationId) {
+        CaseDetails caseDetails = asCaseDetails(caseData);
+        caseDetails.getData().put("migrationId", migrationId);
+        return caseDetails;
+    }
+
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Nested
+    class Dfpl1161 {
```
Comment (so99y) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseControllerTest.java:145]: and DFPL-1161 should be removed?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 4]
```diff
@@ -48,13 +42,16 @@ public class MigrateCaseController extends CallbackController {
...
     private final Map<String, Consumer<CaseDetails>> migrations = Map.of(
-        "DFPL-1144", this::run1144,
-        "DFPL-872rollback", this::run872Rollback,
-        "DFPL-1072", this::run1072,
-        "DFPL-1163", this::run1163,
-        "DFPL-1165", this::run1165,
-        "DFPL-1192", this::run1192,
-        "DFPL-1215", this::run1215
+        "DFPL-1204", this::run1204,
+        "DFPL-1064", this::run1064,
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:46]: should be removed? It had been done.

Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:46]: Imagine there should be four tests should be added and some of them are removed.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 5]
```diff
@@ -42,19 +36,23 @@
...
-        "DFPL-872rollback", this::run872Rollback,
-        "DFPL-1072", this::run1072,
-        "DFPL-1163", this::run1163,
-        "DFPL-1165", this::run1165,
-        "DFPL-1192", this::run1192,
-        "DFPL-1215", this::run1215
+        "DFPL-1204", this::run1204,
+        "DFPL-1202", this::run1202,
+        "DFPL-1195", this::run1195,
+        "DFPL-1065", this::run1065,
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:51]: 1065 should be removed from this list.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 6]
```diff
@@ -42,19 +36,23 @@
...
+        "DFPL-1029", this::run1029,
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:52]: removed!

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 7]
```diff
@@ -42,19 +36,23 @@
...
+        "DFPL-1162", this::run1162,
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:53]: 1029 & 1162 can be removed.

Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:53]: Think you can just simply put your PRs
`"DFPL-1202", this::run1202,
"DFPL-1195", this::run1195,
"DFPL-1204", this::run1204,
"DFPL-1218", this::run1218`

**Review by so99y — DISMISSED** [review]
LGTM

**Review by so99y — DISMISSED** [review]

**Review by so99y — APPROVED** [review]
Looks good. merge after running 1194

**Pull request merged**